src/RequestorPages/UpcomingPosts.js

```diff
@@ -77,19 +94,59 @@ class UpcomingPosts extends Component {
                   <Icon name="add-box" size={30} style={{color:'white'}}/>
                 </TouchableOpacity>
                 <ScrollView style={styles.scrollView}>
-                <View>
-                  {jobsKeys.length > 0 ? (
-                    jobsKeys.map(key => (
-                      <Job
-                        key={key}
-                        id={key}
-                        job={this.state.jobs[key]}
-                      />
-                    ))
-                  ) : (
+                    {this.state.jobs.length > 0 ? (
```

Comment:
Again, why move this logic out of the Job object and into the render method?

Reply:
See response to PastPosts page